packages/core/src/http-middlewares/after/throw-for-status.js

```diff
@@ -1,12 +1,10 @@
 'use strict';
 
-const { stripQueryFromURL } = require('../../tools/http');
+const errors = require('../../errors');
 
 const throwForStatus = resp => {
   if (resp.status > 300) {
-    const cleanURL = stripQueryFromURL(resp.request.url);
-    const message = `Got ${resp.status} calling ${resp.request.method} ${cleanURL}, expected 2xx.`;
-    throw new Error(message);
+    throw new errors.ResponseError(resp);
```

Comment:
(IMPORTANT) As discussed in our call, we should double check we don't introduce a regression of https://github.com/zapier/zapier/issues/32319, where users can see private info of the app in the request query params when an error happens.

Reply:
We only use the URL in https://github.com/zapier/zapier/blob/a69df1a0e98159383489e99547e1e7a2cc30c2bb/web/backend/zapier/errorbox/extract.py#L403-L412 and we probably gonna drop that code so then I could actually update `ResponseError` to not include it in the `JSON.stringify`-ed message at all.

That said, I do like the idea we discussed to update https://github.com/zapier/zapier/blob/a69df1a0e98159383489e99547e1e7a2cc30c2bb/web/backend/zapier/errorbox/extract.py#L205 to use a passed authentication to ensure that we don't extract sensitive stuff from a response body.

However, as this would be a `zapier/zapier` PR and I don't think it needs to be blocking as this PR simply brings v3 on par with v1/2 error extraction.

Reply:
> I don't think it needs to be blocking as this PR simply brings v3 on par with v1/2 error extraction.

I meant we should do a quick test to see if it would create a security issue like https://github.com/zapier/zapier/issues/32319. I just did test with an app that uses a `beforeRequest` middleware that adds the API key to the request URL:

```
const includeApiKey = (request, z, bundle) => {
  if (bundle.authData.api_key) {
    request.params = request.params || {};
    request.params.api_key = bundle.authData.api_key;
  }
  return request;
};
```

and with a trigger that always throws an error:

```
const perform = async (z, bundle) => {
  const response = await z.request({
    url: 'https://auth-json-server.zapier-staging.com/movies',
  });
  throw new z.errors.ResponseError(response);
};
```

When testing in the Editor, the API key isn't in the error message (which is good):

![](https://cdn.zappy.app/3b3e0e416b720afd49212fe0d8b147f3.png)

The part that says `app.com` is weird though, but that doesn't block this PR.

Reply:
Ha, good catch - that is because I'm not actually using the URL in https://github.com/zapier/zapier/blob/a69df1a0e98159383489e99547e1e7a2cc30c2bb/web/backend/zapier/developer_cli/utils.py#L692. I'll fix that.

Reply:
PR: https://github.com/zapier/zapier/pull/38615